packages/orchestrator-ui-components/src/components/WFOWorkflowSteps/WFOStep/WFOStep.tsx

```diff
@@ -49,24 +47,26 @@ export const WFOStep = React.forwardRef(
             stepDurationStyle,
             stepToggleExpandStyle,
             stepRowStyle,
+            euiCodeBlockStyle,
         } = getStyles(theme);
         const t = useTranslations('processes.steps');
-        const hasHtmlMail = delta?.hasOwnProperty('confirmation_mail');
+        const hasHtmlMail = stepDelta?.hasOwnProperty('confirmation_mail');
         const HIDDEN_KEYS = ['label_', 'divider_', '__', 'confirmation_mail'];
 
-        // Todo: refactor into something pure and beautiful
-        let filteredStepDelta = delta;
-        let filteredStepDeltaEmpty = false;
-        if (!showHiddenKeys) {
-            filteredStepDeltaEmpty = true;
-            filteredStepDelta = {};
-            for (const key in delta) {
-                if (!HIDDEN_KEYS.some((word) => key.startsWith(word))) {
-                    filteredStepDeltaEmpty = false;
-                    filteredStepDelta[key] = delta[key];
-                }
-            }
-        }
+        const stepContent = showHiddenKeys
```

Comment:
Personally I would move this into a separate function and probably get rid of the turnary and replace it by if/else.

Too bad we need this functionality to get rid of some 'random' artifacts like label_ and divider. (btw, what happens if you use just label instead of label_1?) I think we need cleaner data in our components.

Reply:
Extracted a function and got rid of the unreadable ternary operator